lib/reek/smells/smell_warning.rb

```diff
@@ -53,12 +55,17 @@ def matches_smell_type?(klass)
         smell_classes.include?(klass.to_s)
       end
 
-      def matches_smell_details?(other_smell_details = {})
-        common_parameters_equal?(other_smell_details)
+      def matches_attributes?(other_attributes = {})
+        check_attributes_comparability(other_attributes)
+        return false unless common_parameters_equal?(other_attributes[:parameters])
+        other_attributes.delete(:parameters)
```

Comment:
that's an interesting idea -  but wouldn't it be cleaner in the long run to just make all the parameter options into instance variables on smell_warning? and then we don't have what seems like an artificial distinction between some aspects of the smell_warning that are instance vars, and others that are collected into a open ended group called "parameters", but as I say that feels like a large refactoring that I don't feel qualified to do ... yet :-)